sycl-jit/jit-compiler/CMakeLists.txt

```diff
@@ -9,25 +9,39 @@ else()
 set(SYCL_JIT_VIRTUAL_TOOLCHAIN_ROOT "/sycl-jit-toolchain/")
 endif()
 
-set(SYCL_JIT_RESOURCE_DEPS
-  sycl-headers    # include/sycl
-  clang           # lib/clang/N/include
-  ${CMAKE_CURRENT_SOURCE_DIR}/utils/generate.py)
+set(SYCL_JIT_RESOURCE_DEPS sycl-headers clang)
+set(SYCL_JIT_RESOURCE_INSTALL_COMPONENTS sycl-headers OpenCL-Headers clang-resource-headers)
 
 if ("libclc" IN_LIST LLVM_ENABLE_PROJECTS)
   # Somehow just "libclc" doesn't build "remangled-*" (and maybe whatever else).
   list(APPEND SYCL_JIT_RESOURCE_DEPS libclc libspirv-builtins) # lib/clc/*.bc
+  list(APPEND SYCL_JIT_RESOURCE_INSTALL_COMPONENTS libspirv-builtins)
 endif()
 
 if ("libdevice" IN_LIST LLVM_ENABLE_PROJECTS)
   list(APPEND SYCL_JIT_RESOURCE_DEPS libsycldevice) # lib/*.bc
+  list(APPEND SYCL_JIT_RESOURCE_INSTALL_COMPONENTS libsycldevice)
 endif()
 
+set(ALL_RTC_PREPARE_RESOURCES_COMMANDS)
+foreach(component IN LISTS SYCL_JIT_RESOURCE_INSTALL_COMPONENTS)
+  list(APPEND ALL_RTC_PREPARE_RESOURCES_COMMANDS
+    COMMAND ${CMAKE_COMMAND} --install ${CMAKE_BINARY_DIR} --prefix ${CMAKE_CURRENT_BINARY_DIR}/rtc-resources-install --component "${component}"
+  )
+endforeach()
+
+add_custom_target(rtc-prepare-resources
```

Comment:
It looks like `add_custom_target` is [always](https://cmake.org/cmake/help/v3.31/command/add_custom_target.html) considered out of date, so this will be run any time someone builds even if it's incremental. Is that a big deal? Is this expensive to run?

Reply:
> Is that a big deal?

Not sure

> Is this expensive to run?

No (at least IMO), and the incremental builds will output this:

```
-- Up-to-date: <root>/build/tools/sycl-jit/jit-compiler/rtc-resources-install/lib/libsycl-fallback-imf-fp64.o
```

instead of
```
-- Installing: <root>/build/tools/sycl-jit/jit-compiler/rtc-resources-install/lib/libsycl-fallback-imf-fp64.o
```

Reply:
Ah I see, it will just call the target which itself will determine it is up to date, so basically nothing happens and it's better than before IMO

Reply:
Updating `sycl.hpp` works with this version (AFAIK), I was thinking about deleting it. But, to be fair, we have that problem everywhere else, so probably no reason to address here.